pkg/endpoint/endpoint.go

```diff
@@ -257,3 +265,16 @@ func networkAndAddress(str string) (string, string) {
 	}
 	return "", parts[0]
 }
+
+// validateStorageEndpoint validates the given string can be used to infer backend and optionally extract connection
+// info
+func validateStorageEndpoint(str string) error {
+	if strings.Contains(str, "://") {
+		return nil
+	}
+	if str == "" || str == "sqlite" {
+		return nil
+	}
```

Comment:
This feels like it's duplicating the check in the `switch network` block above. Can we keep the scheme checking in one place?